### ISSUE TITLE
Name tf.train.Optimizer is deprecated.

### ISSUE THREAD

**Issue by isaac-jennings** [carpentries-incubator/intro-image-classification-cnn#29]
When running compile code:
```
model_intro.compile(optimizer = 'adam',
                    loss = keras.losses.CategoricalCrossentropy(),
                    metrics = ['accuracy'])
```
The following error prints:
`WARNING:tensorflow:From C:\Users\S2996310\AppData\Local\anaconda3\Lib\site-packages\keras\src\optimizers\__init__.py:309: The name tf.train.Optimizer is deprecated. Please use tf.compat.v1.train.Optimizer instead.`

As per [this](https://github.com/keras-team/keras/issues/13326) Keras GitHub issue, `keras-preprocessing` module requires an update.

**Recommend postponing such a fix until such time testing for module conflicts can occur end to end.**

**Comment by isaac-jennings** [comment]
- Only a warning
- Will require fix on researcher side post-fix is applying code from workshop to own use case

Short term resolution: *label wont fix*
Long term resolution: *identify subject matter expert on Python dependency management (software engineer, PhD/post-doc who have contemporary knowledge of this problem or one like it [Danny Meloncelli?])

Isaac to follow up 29/04/2024
